src/pages/Theme/index.tsx

```diff
@@ -1,22 +1,50 @@
+import type { AxiosError } from 'axios';
+import { useQuery } from 'react-query';
 import { Navigate, useParams } from 'react-router-dom';
 
-import { ThemeGoodsSection } from '@/components/features/Theme/ThemeGoodsSection';
-import { getCurrentTheme, ThemeHeroSection } from '@/components/features/Theme/ThemeHeroSection';
+import { Loader } from '@/components/common/Spinner';
+import { ThemeProductsSection } from '@/components/features/Theme/ThemeGoodsSection';
+import { ThemeHeader } from '@/components/features/Theme/ThemeHeroSection';
 import { RouterPath } from '@/routes/path';
-import { ThemeMockList } from '@/types/mock';
+import type { ThemeData } from '@/types';
+import apiClient from '@/utils/api';
+
+const fetchThemeDetails = async (themeKey: string) => {
+  const response = await apiClient.get<{ themes: ThemeData[] }>('/themes');
+  return response.data.themes.find((theme) => theme.key === themeKey);
+};
 
 export const ThemePage = () => {
   const { themeKey = '' } = useParams<{ themeKey: string }>();
-  const currentTheme = getCurrentTheme(themeKey, ThemeMockList);
+
+  const {
+    data: currentTheme,
+    isLoading: isThemeLoading,
+    isError: isThemeError,
+    error,
+  } = useQuery(['theme', themeKey], () => fetchThemeDetails(themeKey), {
```

Comment:
```suggestion
  } = useQuery<ThemeData | undefined, AxiosError>(['theme', themeKey], () => fetchThemeDetails(themeKey), {
```

useQuery에 직접 타입 제공해주면 34번 라인의 `as AxiosError` 제거할 수 있을 거 같네요. `as`를 통한 type assertion은 코드를 개선해야하는 신호로 받아들이시는게 좋습니다. TypeScript 컴파일러에 잘못된 정보를 전달할 수 있으니 꼭 필요한 상황이 아니라면 쓰지 않는게 좋아요.